source/agora/consensus/protocol/Nominator.d

```diff
@@ -932,6 +932,13 @@ extern(D):
                 height, last_height);
             return;  // slot was already externalized or envelope is too new
         }
+        else if (!this.ledger.hasMajoritySignature(last_height))
+        {
+            log.trace("valueExternalized: Will not externalize envelope with slot id {} as we are " ~
+                "missing signagures for height {}", height, last_height);
+            return;
+        }
```

Comment:
I'm confused, should we even reach `valueExternalized` if we can't externalize ?

Reply:
Here is where we `validateValue` but do not block `ValueExternalized`:
https://github.com/bosagora/agora/blob/4269bf72d6b146a24d70a518bd8fa7006dcec9fa/source/agora/consensus/protocol/Nominator.d#L900-L903

Reply:
`validateValue` only represents a single node's view on a value, for now we have a homogenous network but a value might be "invalid" for a single node yet be externalized by the rest of the network.